extensions/consul-config/runtime/src/main/java/io/quarkus/consul/config/runtime/VertxConsulConfigGateway.java

```diff
@@ -0,0 +1,167 @@
+            Optional<String> storePassword, T store) throws Exception {
+        return store
+                .setPassword(storePassword.orElse(""))
+                .setValue(io.vertx.core.buffer.Buffer.buffer(storeBytes(storePath)));
+    }
+
+    private static String determineStoreType(Path keyStorePath) {
+        String pathName = keyStorePath.toString().toLowerCase();
+        if (pathName.endsWith(".p12") || pathName.endsWith(".pkcs12") || pathName.endsWith(".pfx")) {
+            return PKS_12;
+        }
+        return JKS;
+    }
+
+    private static byte[] storeBytes(Path keyStorePath)
+            throws Exception {
+        InputStream classPathResource = Thread.currentThread().getContextClassLoader()
+                .getResourceAsStream(keyStorePath.toString());
+        if (classPathResource != null) {
+            try (InputStream is = classPathResource) {
+                return allBytes(is);
+            }
+        } else {
+            try (InputStream is = Files.newInputStream(keyStorePath)) {
+                return allBytes(is);
+            }
+        }
+    }
+
+    private static byte[] allBytes(InputStream inputStream) throws Exception {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        int nRead;
+        byte[] data = new byte[1024];
+        while ((nRead = inputStream.read(data, 0, data.length)) != -1) {
+            buffer.write(data, 0, nRead);
+        }
+        buffer.flush();
+        return buffer.toByteArray();
+    }
+
+    @Override
+    public Uni<Response> getValue(String key) {
+        HttpRequest<Buffer> request = webClient
+                .get(consulConfig.agent.hostPort.getPort(), consulConfig.agent.hostPort.getHostString(), "/v1/kv/" + key)
+                .ssl(consulConfig.agent.useHttps)
+                .putHeader("Accept", "application/json");
+        if (consulConfig.agent.token.isPresent()) {
+            request.putHeader("Authorization", "Bearer " + consulConfig.agent.token.get());
+        }
+
+        log.debug("Attempting to look up value of key '" + key + "' from Consul.");
+        return request.send().map(r -> {
+            if (r.statusCode() != 200) {
+                log.debug("Look up of key '" + key + "' from Consul yielded a non success HTTP error-code: " + r.statusCode());
+                return null;
+            } else {
+                JsonArray jsonArray = r.bodyAsJsonArray();
+                if (jsonArray.size() != 1) {
+                    throw new IllegalStateException(
+                            "Consul returned an unexpected number of results when looking up value of key '" + key + "'");
+                }
+                JsonObject jsonObject = jsonArray.getJsonObject(0);
+                return new Response(jsonObject.getString("Key"), jsonObject.getString("Value"));
+            }
+        });
+    }
+
+    @Override
+    public void close() {
+        this.webClient.close();
+        this.vertx.close();
```

Comment:
isn't `close` returning a Uni? If so... this is not doing anything.
Use `closeAndForget()` or `closeAndAwait()`